PluralKit.Bot/Commands/SystemEdit.cs

```diff
@@ -523,9 +533,44 @@ public async Task Delete(Context ctx, PKSystem target)
             throw new PKError(
                 $"System deletion cancelled. Note that you must reply with your system ID (`{target.Hid}`) *verbatim*.");
 
-        await ctx.Repository.DeleteSystem(target.Id);
+        // If the user confirms the deletion, export their system and send them the export file before actually
+        // deleting their system, unless they specifically tell us not to do an export.
+
+        var noExport = ctx.MatchFlag("ne", "no-export");
+        if (!noExport)
+        {
+            var json = await ctx.BusyIndicator(async () =>
+            {
+                // Make the actual data file
+                var data = await _dataFiles.ExportSystem(ctx.System);
+                return JsonConvert.SerializeObject(data, Formatting.None);
+            });
+
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes(json));
+
+            try
+            {
+                var dm = await _dmCache.GetOrCreateDmChannel(ctx.Author.Id);
+                var msg = await ctx.Rest.CreateMessage(dm,
+                    new MessageRequest { Content = $"{Emojis.Success} System deleted. If you want to set up your PluralKit system again, you can import the file below with `pk;import`." },
+                    new[] { new MultipartFile("system.json", stream, null) });
+                await ctx.Rest.CreateMessage(dm, new MessageRequest { Content = $"<{msg.Attachments[0].Url}>" });
+
+                // If the original message wasn't posted in DMs, send a public reminder
+                if (ctx.Channel.Type != Channel.ChannelType.Dm)
+                    await ctx.Reply($"{Emojis.Success} System deleted. Check your DMs for a copy of your system's data!");
+            }
+            catch (ForbiddenException)
+            {
+                // If user has DMs closed, tell 'em to open them
+                throw new PKError(
+                    $"Could not send a copy of your system's data in DMs. Are your DMs closed?");
+            }
+        }
+
+        // Now that we've sent the export data (or been told not to), we can safely delete the system
 
-        await ctx.Reply($"{Emojis.Success} System deleted.");
```

Comment:
If noexport is on then it won't send a confirm message, maybe put this in an else